src/components/Warehouse/AddWarehouse.js

```diff
@@ -0,0 +1,302 @@
+      !emailRegex.test(inputValues.contact_email)
+    ) {
+      errors.contact_email = "Please enter a valid email address";
+    }
+
+    // Validate phone number format only if the phone field is not empty
+    if (
+      inputValues.contact_phone &&
+      !phoneRegex.test(inputValues.contact_phone)
+    ) {
+      errors.contact_phone = "Please enter a valid 10-digit phone number";
+    }
+
+    console.log(errors);
+    return errors;
+  };
+
+  //Function to handle form submission
+  const handleSubmit = async (e) => {
+    try {
+      e.preventDefault();
+      setErrors(validateForm(formData));
+      await axios.post("http://localhost:8080/api/warehouses", formData);
+      navigate(`/warehouses`);
+    } catch (error) {
+      console.error("Error posting Warehouse data:", error);
+    }
+  };
+
+  return (
+    <>
+      <Header activeNavItem="warehouses" />
+      <div className="card-container">
+        <h2 className="card-container__heading">
+          <img
+            src={BackIcon}
+            alt="Back Icon"
+            className="inventory-card__icon"
+            onClick={() => {
+              navigate(`/warehouses`);
+            }}
+          />
+          Add New Warehouse
+        </h2>
+        <hr className="inventory-card__divider"></hr>
+        <form>
+          <div className="add-warehouse-container">
+            <div className="add-warehouse-container__column-1">
+              <h2 className="form-heading-add-warehouse">Warehouse Details</h2>
+
+              <label className="form-label-add-warehouse">Warehouse Name</label>
+              <input
+                type="text"
+                name="warehouse_name"
+                value={formData.warehouse_name}
+                onChange={handleChange}
+                placeholder="Warehouse Name"
+                className="form-input-add-warehouse"
+              />
+              {errors.warehouse_name ? (
+                <p className="form-error">
+                  <img
+                    src={ErrorIcon}
+                    alt="Error Icon"
+                    className="form-error__icon"
+                  />
+                  {` ${errors.warehouse_name}`}
+                </p>
+              ) : null}
+              <br></br>
+              <label className="form-label-add-warehouse">Street Address</label>
+              <input
+                type="text"
+                name="address"
+                value={formData.address}
+                onChange={handleChange}
+                placeholder="Street Address"
+                className="form-input-add-warehouse"
+              />
+              {errors.address ? (
+                <p className="form-error">
+                  <img
+                    src={ErrorIcon}
+                    alt="Error Icon"
+                    className="form-error__icon"
+                  />
+                  {` ${errors.address}`}
+                </p>
+              ) : null}
+              <br></br>
```

Comment:
This error handling is very redundant. You should just loop through the `errors` and dynamically fill the JSX.